### ISSUE TITLE
URL_PREFIX is not working as expected

### ISSUE THREAD

**Issue by sebastienvermeille** [WeblateOrg/weblate#2986]
**Describe the bug**
We are using weblate behind a reverse proxy which provides https and basically a prefix.
Example of desired url:  portal.company.com/weblate

I installed weblate on a machine with docker-compose and I added URL_PREFIX=weblate in settings-override.py but unforunately it seems to not work as expected...

If I go to the ip:8999/weblate it redirects me to ip:8999/accounts/login/?next=/weblate with a 404 error page most of the time... 

**Expected behavior**
When I setup the URL_PREFIX weblate use this prefix everywhere and I can simply reach it.

Using docker-compose:

```
 * Weblate 3.8
 * Python 3.7.3
 * Django 2.2.4
 * Celery 4.3.0
 * celery-batches 0.2
 * six 1.12.0
 * social-auth-core 3.2.0
 * social-auth-app-django 3.1.0
 * django-appconf 1.0.3
 * translate-toolkit 2.4.0
 * translation-finder 1.6
 * Whoosh 2.7.4
 * defusedxml 0.6.0
 * Git 2.20.1
 * Pillow 5.4.1
 * python-dateutil 2.8.0
 * lxml 4.3.2
 * django-crispy-forms 1.7.2
 * django_compressor 2.3
 * djangorestframework 3.10.2
 * user-agents 2.0
 * jellyfish 0.7.2
 * diff-match-patch 20181111
 * ruamel.yaml 0.16.2
 * tesserocr 2.4.0
 * Mercurial 4.8.2
 * git-svn 2.20.1
 * Database backends: django.db.backends.postgresql
 * Cache backends: avatar:FileBasedCache, default:RedisCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: 127.0.0.1
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 4.9.0-9-amd64 (x86_64)
```

Any help is welcome... I searched a bit and have seen some people edit python files directly but with docker I have  no idea how to deal with that to be fully honest

**Comment by nijel** [comment]
The problem is that this setting is used in others, so to make it work you would have to copy all settings from example one which use it. The proper fix is to set WEBLATE_URL_PREFIX environment variable for this. I can't find it in the docs now, so apparently documentation is missing here....

29. srpna 2019 17:07:32 SELČ, Sebastien Vermeille <notifications@github.com> napsal:
>**Describe the bug**
>We are using weblate behind a reverse proxy which provides https and
>basically a prefix.
>Example of desired url:  portal.company.com/weblate
>
>I installed weblate on a machine with docker-compose and I added
>URL_PREFIX=weblate in settings-override.py but unforunately it seems to
>not work as expected...
>
>If I go to the ip:8999/weblate it redirects me to
>ip:8999/accounts/login/?next=/weblate with a 404 error page most of the
>time... 
>
>**Expected behavior**
>When I setup the URL_PREFIX weblate use this prefix everywhere and I
>can simply reach it.
>
>
>Using docker-compose:
>
>```
> * Weblate 3.8
> * Python 3.7.3
> * Django 2.2.4
> * Celery 4.3.0
> * celery-batches 0.2
> * six 1.12.0
> * social-auth-core 3.2.0
> * social-auth-app-django 3.1.0
> * django-appconf 1.0.3
> * translate-toolkit 2.4.0
> * translation-finder 1.6
> * Whoosh 2.7.4
> * defusedxml 0.6.0
> * Git 2.20.1
> * Pillow 5.4.1
> * python-dateutil 2.8.0
> * lxml 4.3.2
> * django-crispy-forms 1.7.2
> * django_compressor 2.3
> * djangorestframework 3.10.2
> * user-agents 2.0
> * jellyfish 0.7.2
> * diff-match-patch 20181111
> * ruamel.yaml 0.16.2
> * tesserocr 2.4.0
> * Mercurial 4.8.2
> * git-svn 2.20.1
> * Database backends: django.db.backends.postgresql
> * Cache backends: avatar:FileBasedCache, default:RedisCache
> * Email setup: django.core.mail.backends.smtp.EmailBackend: 127.0.0.1
> * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
> * Platform: Linux 4.9.0-9-amd64 (x86_64)
>```
>
>Any help is welcome... I searched a bit and have seen some people edit
>python files directly but with docker I have  no idea how to deal with
>that to be fully honest

**Comment by nijel** [comment]
Thank you for your report, the issue you have reported has just been fixed. 

* In case you see problem with the fix, please comment on this issue. 
* In case you see similar problem, please open separate issue.
* If you are happy with the outcome, consider supporting Weblate by [donating](https://weblate.org/donate/).

**Comment by sebastienvermeille** [comment]
@nijel thank you for your answer and fix but unfortunately I still get some issues regarding it.

I adapted my docker-compose.override.yml adding : `- WEBLATE_URL_PREFIX=translations` under environment section. After I restarted the whole setup (docker-compose down, docker-compose up -d)

I get a 404 page not found when I access it via /translations and logo, css, js files are not loaded correctly:
```
http://ip:8999/translations/translations/accounts/login/?next=/translations/
http://ip:8999/translations/translations/accounts/login/translations/static/logo-128.png
```
it looks like it doubles the prefix url in some area.

Can I help somehow ?
